storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -363,6 +360,21 @@ func (s *peersSource) incrementalFlush(
 			// A block error is more interesting to bubble up than a close error
 			err = blockErr
 		}
+
+		if seriesCachePolicy != series.CacheAll && seriesCachePolicy != series.CacheAllMetadata {
+			// If we're not going to keep all of the data, or at least all of the metadata in memory
```

Comment:
can this be done asynchronously?

Reply:
I think we probably could but it might complicate the logic a lot. Basically immediately after this executes, this shard result will be merged into the bootstrap result, and at that point messing with it could lead to lots of concurrency issues.

That said we spoke offline and this P.R results in a huge improvement to peer bootstrapping performance in general, so whatever penalty we're paying here shouldn't be an issue